test/matrix_generator.cc

```diff
@@ -1836,79 +1898,43 @@ void generate_matrix(
             #pragma omp parallel
             #pragma omp master
             {
```

Comment:
The below change is just merging the lower and upper cases into one code using `i_start` and `i_end` loop bounds. I did the same in the new zerocol block later, so fixed it here, too.